### DISCUSSION TITLE
How do i get the current Cameras position?

### DISCUSSION THREAD

**Discussion by MCVenor** [stride3d/stride#1936] [Newcomers] [answered]
I try to make a Raycast from the current Viewpoint, the Camera, against a Plane (Plane class).
I wonder how to get the current position of the current Camera.

At this moment im going to use a reference throu properties, but i wonder if there is another way.

**Comment by Doprez** [comment]
You can access it from the GraphicsCompositor. The only hiccup I found is that the camera doesnt populate here until a few frames in and will be null until initialized.
```
	public static CameraComponent GetCamera(this ScriptComponent scriptComponent)
	{
		SceneCameraSlotCollection cameraCollection = scriptComponent.SceneSystem.GraphicsCompositor.Cameras;
		foreach (var sceneCamera in cameraCollection)
		{
			if (sceneCamera.Name == "Main")
			{
				return sceneCamera.Camera;
			}
		}
		return null;
	}
```
the world position can then be accessed with the entity world matrix `camVar.Entity.Transform.WorldMatrix.TranslationVector;`

**Reply by MCVenor** [reply]
🔥

**Comment by MCVenor** [comment]
Holy cow thats cool. 
But my spider-it-senses tell me not to do it this way :D

**Reply by Doprez** [reply, marked as answer]
When it comes to finding the camera with the main tag it would have to be from the GraphicsCompositor.
`scriptComponent.SceneSystem.GraphicsCompositor.Cameras;` 
The GraphicsCompositor contains the CameraSlots that you would assign on the Camera itself with the Slot property. This is why I am looking for the `sceneCamera.Name == "Main"`.

if you wanted to just find the camera in the current scene then you could do a search for an Entity with a specific name using SceneSystem, below would be the LINQ version.
`SceneSystem.SceneInstance.RootScene.Entities.Where(x => x.Name == "NameOfCameraEntity");`

it does feel funky but this is essentially the same idea as Unity's Camera.Main for the initial code I put.

**Reply by MCVenor** [reply]
Yes, my initional thought was that we use a GraphicsCompositor to access what feels like should be accessed otherwise. But your explanation is sound. If i had to loop thou ALL stuff vs SOME Cameras, the later is indeed better. Thanks for the awnser

**Reply by Ethereal77** [reply]
If you control the camera in the scene in which the ray-casting script is (for example, it's a isometric camera, and the script is the one responsible for scrolling, picking, etc), it's better to assign manually the camera to a property in the script. That way you know it's the correct camera and you don't have to go enumerating entities or accessing the `GraphicsCompositor.Cameras` comparing names (which is brittle and error-prone). Like this:
```cs
class MyCameraScript : SyncScript
{
    public CameraComponent IsoCamera; // Or its owner Entity. Can't remember if you can drag-drop components directly
}
```
Then, as you are composing the scene, assign the camera component (or entity) to the `MyCameraScript` in the properties panel.
